docs/build/isc/v1.1/docs/how-tos/core-contracts/nft/get-L2-nfts.mdx

```diff
@@ -0,0 +1,97 @@
+
+# Get NFTs Owned by an Account
+
+This guide will show you how to get the L2 NFTs owned by an account using the [`ISCAccounts.getL2NFTs`](../../../reference/magic-contract/ISCAccounts.md#getl2nfts) function from the [`ISCAccounts.getL2NFTs`](../../../reference/magic-contract/ISCAccounts.md) [magic contract](../../../reference/magic-contract/introduction.md).
+
+<GetNftMetadata />
+
+## Understanding the `getL2NFTs` Function
+
+The `getL2NFTs` function is a view function provided by the `ISCAccounts` interface. It takes an `ISCAgentID` as an argument and returns an array of `NFTID` objects. This function allows users to query and retrieve the list of L2 NFTs owned by a specified agent. The `ISCAgentID` represents the identifier of the agent (user or contract) whose NFTs you want to retrieve. The returned array contains the identifiers of the NFTs, which can then be used to fetch more details about each NFT if needed.
+
+## Why Would a User Need to Retrieve Owned NFTs?
+
+Retrieving the NFTs owned by a user can serve several purposes:
+
+- ### Portfolio Management
+
+  - Users can manage their digital assets effectively by viewing all their NFTs in one place.
+
+- ### Marketplace Listing
+
+  - Users can list their NFTs on various marketplaces by verifying their ownership.
+
+- ### Collection Display
+
+  - Users can showcase their NFT collections on social media, personal websites, or within specific applications.
+
+- ### Trading and Transfers
+
+  - Knowing the NFTs owned helps users decide which assets to trade or transfer to other users.
+
+- ### Application Integration
+
+  - Applications can provide customized experiences based on the NFTs a user owns, such as granting access to special content or features.
+
+## Prerequisites
+
+### 1. Install the ISC Magic Library
+
+   Ensure you have the ISC Magic library installed in your project:
+
+   ```bash
+   npm install @iota/iscmagic
+   ```
+
+### 2. Import the ISC Library
+
+   In your Solidity file, import the ISC Library:
+
+   ```solidity
+   pragma solidity >=0.8.24;
+
+   import "@iota/iscmagic/ISC.sol";
+   ```
+
+## Implement the `getL2NFTs` Function
+
+Here’s a sample implementation to retrieve L2 NFTs owned by an account:
+
+```solidity
+function getOwnedNFTs(ISCAgentID memory agentID) public view returns (NFTID[] memory) {
+    // Call the getL2NFTs function from the ISCAccounts contract
+    NFTID[] memory ownedNFTs = ISC.accounts.getL2NFTs(agenID);
+    return ownedNFTs;
+}
+```
+
+## Full Example Contract
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import "@iota/iscmagic/ISC.sol";
+
+contract MyNFTContract {
+
+function getOwnedNFTs(ISCAgentID memory agentID) public view returns (NFTID[] memory) {
+    // Call the getL2NFTs function from the ISCAccounts contract
+    NFTID[] memory ownedNFTs = ISC.accounts.getL2NFTs(agenID);
+    return ownedNFTs;
+    }
+}
+```
+## Conclusion
```

Comment:
I like consistency, so I would remove it as we don't have it in other how-tos either. But no strong opinion from my side. Question is if someone will read it after the full how-to